network/proto/handlers.go

```diff
@@ -170,34 +208,74 @@ func (p *protocol) checkTransactionOnLocalNode(peer p2p.PeerID, transactionRef h
 		// TODO: Currently we send the query to the peer that sent us the hash, but this peer might not have the
 		//   transaction contents. We need a smarter way to get it from a peer who does.
 		log.Logger().Infof("Received transaction hash from peer that we don't have yet or we're missing its contents, will query it (peer=%s,hash=%s)", peer, transactionRef)
-		responseMsg := createMessage()
-		responseMsg.Message = &transport.NetworkMessage_TransactionPayloadQuery{
-			TransactionPayloadQuery: &transport.TransactionPayloadQuery{PayloadHash: transaction.PayloadHash().Slice()},
-		}
-		return p.p2pNetwork.Send(peer, &responseMsg)
+		p.sender.sendTransactionPayloadQuery(peer, transaction.PayloadHash())
 	}
 	return nil
 }
 
-func (p *protocol) handleTransactionListQuery(peer p2p.PeerID) error {
-	log.Logger().Tracef("Received transaction list query from peer (peer=%s)", peer)
-	transactions, err := p.graph.All()
+func (p *protocol) handleTransactionListQuery(peer p2p.PeerID, blockDateInt uint32) error {
+	var startDate time.Time
+	var endDate time.Time
+	if blockDateInt == 0 {
+		logrus.Warnf("Peer didn't specify date of the block to be queried, ignoring (peer=%s).", peer)
+		// Historic block is queried, query from start up to the first block
+		// endDate = p.blocks.Get()[1].Start
+		return nil
+	}
+	startDate = time.Unix(int64(blockDateInt), 0).UTC()
```

Comment:
I removed all `UTC` conversions; `time.Unix()` and `Time.Unix()` always use UTC (leading to the same results), regardless of the `Location` of the `Time` struct.

The spec states it must be Unix timestamp in seconds since epoch in UTC, so there can be no confusion on that side.